src/openscad.cc

```diff
@@ -732,6 +732,20 @@ void registerDefaultIcon(QString applicationFilePath) {
 void registerDefaultIcon(const QString&) { }
 #endif
 
+static void assertValidPath()
+{
+#ifndef __WIN32
+  boost::system::error_code ec{};
+  auto current_path = fs::current_path(ec);
+
+  while(ec.value() != 0)
```

Comment:
How safe is it to assume it will always reach a state where it exits the loop?

Reply:
In a UNIX like system it should always reach at least `/`. I tested this on my Linux box, and couldn't figure out a scenario where / would not be reached.  I'd like to test on macOS, but right now I have only a  Linux machine available.

On Windows I don't think it is possible ensure reaching an existing directory. Nor am I sure that reproducing the issue is even possible, as I remember, if you have Explorer of terminal in a directory, it is not possible to remove it.